schematic/schemas/json_schema_generator.py

```diff
@@ -0,0 +1,747 @@
+            [
+                node_display_name in node_processor.reverse_dependencies,
+                is_node_required,
+                node_processor.current_node in node_processor.root_dependencies,
+            ]
+        )
+        if set_property:
+            # Determine if current node has conditional dependencies that need to be set
+            if node_display_name in node_processor.reverse_dependencies:
+                _set_conditional_dependencies(
+                    json_schema,
+                    node_processor.current_node,
+                    node_display_name,
+                    node_processor.reverse_dependencies,
+                    node_processor.range_domain_map,
+                )
+                is_node_required = False
+            _set_property(
+                json_schema,
+                node_processor.current_node,
+                node_range_display_names,
+                node_validation_rules,
+                is_node_required,
+                node_description,
+            )
+            node_processor.update_processed_nodes_with_current_node()
+
+        # add process node as a conditional to its dependencies
+        node_dependencies_d = self.dmge.get_nodes_display_names(
+            node_list=node_dependencies
+        )
+        node_processor.update_reverse_dependencies(
+            node_display_name, node_dependencies_d
+        )
+        # add nodes found as dependencies and range of this processed node
+        # to the list of nodes to be processed
+        node_processor.update_nodes_to_process(sorted(node_range))
+        node_processor.update_nodes_to_process(sorted(node_dependencies))
+
+
+def _write_data_model(
+    json_schema_dict: dict[str, Any],
+    schema_path: Union[str, None],
+    name: str,
+    jsonld_path: str,
+) -> None:
+    """
+    Creates the JSON Schema file
+
+    Arguments:
+        json_schema_dict: The JSON schema in dict form
+        schema_path: Where to save the JSON Schema file
+        jsonld_path:
+          The path to the JSONLD model, used to create the path
+          Used if schema_path is None
+        name:
+          The name of the datatype(source node) the schema is being created for
+          Used if schema_path is None
+    """
+    if schema_path:
+        json_schema_path = schema_path
+    if not schema_path:
+        json_schema_path = get_json_schema_log_file_path(
+            data_model_path=jsonld_path, source_node=name
+        )
+        json_schema_dirname = os.path.dirname(json_schema_path)
+        if json_schema_dirname != "":
+            os.makedirs(json_schema_dirname, exist_ok=True)
+
+        logger.info(
+            "The JSON schema file can be inspected by setting the following "
+            "nested key in the configuration: (model > location)."
+        )
+    export_json(json_doc=json_schema_dict, file_path=json_schema_path, indent=2)
+
+
+def _set_conditional_dependencies(
+    json_schema: JSONSchema,
```

Comment:
Same note: https://github.com/Sage-Bionetworks/schematic/pull/1604/files#r2089909192

Reply:
@BryanFauble apologies, that link doesn't seem to lead anywhere. Is it a resolved comment?

Reply:
> This is a good candidate to be a method on the JSONSchema class. Usually it's bad practice to update the values of an argument that you pass into a function